lib/backend/kubernetes/kubernetes.go

```diff
@@ -191,15 +194,23 @@ func NewSharedWithClient(restClient kubernetes.Interface) (*Backend, error) {
 }
 
 // NewWithConfig returns a new instance of Kubernetes Secret identity backend storage with the provided config.
-func NewWithConfig(conf Config) (*Backend, error) {
+// Ensures that the secret exists.
+func NewWithConfig(ctx context.Context, conf Config) (*Backend, error) {
 	if err := conf.Check(); err != nil {
 		return nil, trace.Wrap(err)
 	}
-
-	return &Backend{
+	b := &Backend{
 		Config: conf,
-		mu:     sync.Mutex{},
-	}, nil
+	}
+	if b.Exists(ctx) {
+		return b, nil
+	}
+	// NOTE: This will trigger an actual Create() since we don't have a resource version set.
+	// If we have an AlreadyExists error, discard it and move on as it can happen with rapid restarts/rollouts/agent name collisions.
+	if err := b.upsertSecret(ctx, b.genSecretObject()); err != nil && !kubeerrors.IsAlreadyExists(trace.Unwrap(err)) {
```

Comment:
We should change the operation to create*

In our case, using upsert doesn’t make sense-if the secret has been deleted, the process should fail, since critical details are missing and we should avoid levering it. On the next restart, it would likely fail anyway because half of the keys are missing

I propose having only two operations:

* **create** – creates the secret only if it does not already exist.
* **update** – updates the secret’s contents when modifying details.

Reply:
Done.